corehq/motech/fhir/tasks.py

```diff
@@ -394,5 +398,252 @@ def create_parent_indices(
     )
 
 
+def generate_epic_jwt():
+    key = settings.EPIC_PRIVATE_KEY
+    # token will expire in 4 mins
+    exp = int(time.time()) + 240
+    jti = str(uuid.uuid4())
+    header = {
+        "alg": "RS256",
+        "typ": "JWT",
+    }
+    payload = {
+        "iss": settings.EPIC_CLIENT_ID,
+        "sub": settings.EPIC_CLIENT_ID,
+        "aud": "https://fhir.epic.com/interconnect-fhir-oauth/oauth2/token",
+        "jti": jti,
+        "exp": exp
+    }
+    token = jwt.encode(payload, key, algorithm="RS256", headers=header)
+    return token
+
+
+def request_epic_access_token():
+    headers = {
+        "Content_Type": "application/x-www-form-urlencoded",
+    }
+    data = {
+        "grant_type": "client_credentials",
+        "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+        "client_assertion": generate_epic_jwt()
+    }
+    url = "https://fhir.epic.com/interconnect-fhir-oauth/oauth2/token"
+    response = requests.post(url, data=data, headers=headers)
+    if response.status_code == 200:
+        return response.json().get('access_token')
+    elif response.status_code >= 400:
+        return response.raise_for_status()
+
+
+def get_patient_fhir_id(given_name, family_name, birthdate, access_token):
+    url = f"https://fhir.epic.com/interconnect-fhir-oauth/api/FHIR/R4/Patient?birthdate={birthdate}&family={family_name}&given={given_name}&_format=json"
+    headers = {
+        'authorization': 'Bearer %s' % access_token,
+    }
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        response_json = response.json()
+        fhir_id = None
+        entry = response_json.get('entry')[0]
+        if entry: 
+            resource = entry.get('resource')
+            if resource:
+                fhir_id = resource.get('id')
+        return fhir_id
+    elif response.status_code >= 400:
+        response.raise_for_status()
+
+
+# TODO add time param 12 weeks from study start date
+def get_epic_appointments_for_patient(fhir_id, access_token):
+    appointments = []
+    headers = {
+        'authorization': 'Bearer %s' % access_token,
+    }
+    url = f"https://fhir.epic.com/interconnect-fhir-oauth/api/FHIR/R4/Appointment?&patient={fhir_id}&service-category=appointment&_format=json"
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        json_response = response.json()
+        entries = json_response['entry']
+        for entry in entries:
+            appointments.append(entry)
+    elif response.status_code >= 400:
+        response.raise_for_status()
+    return appointments
+
+
+def convert_date_and_time_to_utc_timestamp(date, time):
+    date_time = date + "T" + time
+    utc_zone = dateutil.tz.gettz('UTC')
```

Comment:
nit: you can use the UTC constant: https://docs.python.org/3/library/datetime.html#datetime.timezone.utc
```suggestion
    utc_zone = timezone.utc
```

Reply:
changed here: https://github.com/dimagi/commcare-hq/pull/35125/commits/67fae622d9b7b5d714bfa0bc901dab61a993447a